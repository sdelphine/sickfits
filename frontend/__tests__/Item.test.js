import Item from '../components/Item';
import { shallow, mount } from 'enzyme';
import PriceTag from '../components/styles/PriceTag'
import Title from '../components/styles/Title';
import ItemStyles from '../components/styles/ItemStyles';
import DeleteItem from '../components/DeleteItem'
import toJSON from 'enzyme-to-json'


const fakeItem = {
    id: 'ABC123',
    title: 'A Cool Item',
    price: 5000,
    description: 'This item is really cool!',
    image: 'dog.jpg',
    largeIamge: 'largedog.jpg',
}

describe('<Item />', () => {
    it('renders and matches the snapshot', () => {
        // const price = '$50.35'
        // expect(price).toMatchSnapshot()

        const wrapper = shallow(<Item item={fakeItem} />)
        expect(toJSON(wrapper)).toMatchSnapshot()

    })

    // it('renders the image properly', () => {
    //     const wrapper = shallow(<Item item={fakeItem} />)
    //     const image = wrapper.find('img').props()
    //     // console.log(image)
    //     expect(image.src).toBe(fakeItem.image)
    //     expect(image.alt).toBe(fakeItem.title)
    // })

    // it('renders the priceTag and title properly', () => {
    //     const wrapper = shallow(<Item item={fakeItem} />)
    //     // console.log(wrapper.debug())
    //     const priceTag = wrapper.find(PriceTag).children().text()
    //     // console.log(priceTag)
    //     expect(priceTag).toBe('$50')
    //     const title = wrapper.find(Title).find('a').children().text()
    //     expect(title).toBe(fakeItem.title) 
    // })

    // it('renders out the buttons properly', () => {
    //     const wrapper = shallow(<Item item={fakeItem} />)
    //     const buttonList = wrapper.find(ItemStyles).find('.buttonList')
    //     expect(buttonList.children()).toHaveLength(3)
        
        
    //     expect(buttonList.find('Link').exists()).toBe(true)
    //     // expect(buttonList.find('Link')).toHaveLength(1)
    //     // expect(buttonList.find('Link').exists()).toBeTruthy()
        
    //     // AddToCart Not implemented yet 
    //     // expect(buttonList.find(AddToCart).exists()).toBe(true)

    //     expect(buttonList.find(DeleteItem).exists()).toBe(true)

    // })
})