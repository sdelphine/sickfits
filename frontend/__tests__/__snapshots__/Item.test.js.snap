// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ForwardRef(ItemStyles__Item)>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <ForwardRef(Title)>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </ForwardRef(Title)>
  <ForwardRef(PriceTag)>
    $50
  </ForwardRef(PriceTag)>
  <p>
    This item is really cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ‚úè 
      </a>
    </Link>
    <button>
      Add To Cart
    </button>
    <DeleteItem
      id="ABC123"
    >
      Delete this item
    </DeleteItem>
  </div>
</ForwardRef(ItemStyles__Item)>
`;
